### PR TITLE
Enhancement/6520 button tooltip delay

### PR DESCRIPTION
**Pull request by zutigrm**
## Summary

Addresses issue:

- #6520 

## Relevant technical choices

enterDelay prop is not introduced directly in tooltip per IB, since tooltip component accepts spread props, so there was no need for it to be added explicitly. The button prop is updated. I also added tooltip delay on menu items not mentioned in IB, since these options probably arrived later after this issue was opened - so included are dashboard sharing menu item, as well as view only menu item from user with view only access.

## PR Author Checklist

- [x] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 5.2 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

---------------

_Do not alter or remove anything below. The following sections will be managed by moderators only._

## Code Reviewer Checklist

- [x] Run the code.
- [x] Ensure the acceptance criteria are satisfied.
- [x] Reassess the implementation with the IB.
- [x] Ensure no unrelated changes are included.
- [x] Ensure CI checks pass.
- [x] Check Storybook where applicable.
- [x] Ensure there is a QA Brief.

## Merge Reviewer Checklist

- [x] Ensure the PR has the correct target branch.
- [x] Double-check that the PR is okay to be merged.
- [x] Ensure the corresponding issue has a ZenHub release assigned.
- [x] Add a changelog message to the issue.

### CODE REVIEW

File: assets/js/googlesitekit/components-gm2/Button.js [hunk 1]
```diff
@@ -120,7 +121,14 @@ const Button = forwardRef(
 			( icon && tooltipTitle && children === undefined )
 		) {
 			return (
-				<Tooltip title={ tooltipTitle }>{ ButtonComponent }</Tooltip>
+				<Tooltip
+					title={ tooltipTitle }
+					{ ...( tooltipEnterDelayInMS !== undefined
+						? { enterDelay: tooltipEnterDelayInMS }
+						: {} ) }
```
Comment (nfmohit) [inline assets/js/googlesitekit/components-gm2/Button.js:128]: While this works and is absolutely valid, it doesn't look like we use this conditional format anywhere else. Also, in my opinion, it is slightly less readable.

What do you think about setting a default value of the `tooltipEnterDelayInMS` prop to `100` (that seems to be [the default value in the API](https://mui.com/material-ui/api/tooltip/#Tooltip-prop-enterDelay)) and passing `enterDelay` unconditionally to `<Tooltip>` component?

```suggestion
					enterDelay: tooltipEnterDelayInMS
```

**Review by nfmohit — CHANGES_REQUESTED** [review]
Hi @zutigrm.
Thank you for your work on this.

I have pointed out a nitpicky concern in this review. Apart from that, I have a couple more observations:

1. It looks like the implementation only leaves out the `EntitySearchInput` component. I understand it is not mentioned in the ACs, but would you mind including that as well for consistency, similar to how you've done the same for dashboard sharing?
2. Would you be so kind as to update the QAB to clarify that no delay should be expected in seeing a tooltip when hovering over the next item right away?

Please let me know what you think. Thank you!

File: assets/js/googlesitekit/components-gm2/Button.js (hunk 1 above)
Comment (zutigrm) [reply assets/js/googlesitekit/components-gm2/Button.js]: @nfmohit Thanks, that makes sense

**Review by nfmohit — APPROVED** [review]
LGTM, thank you @zutigrm !

**Pull request merged**